src/components/dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState } from "react"
+import "./dropdown.scss"
+import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+
+
+export enum DropdownSelectType {
+    CHECKBOX = 'CHECKBOX',
+    LIST = 'LIST',
+}
+export interface DropdownProps {
+    options: { label: string, value: string }[];
+    defaultText: string;
+    type?: DropdownSelectType;
+    boxStyle?: {};
+    listStyle?: {};
+    disabled?: boolean;
+}
+
+const Dropdown = (props: DropdownProps) => {
```

Comment:
What's missing here is a full stateful clone of options that maintains the state of whether or not each option is selected. That should allow you to move the document interactions to state vs document interactions.